packages/preferences/src/browser/abstract-resource-preference-provider.ts

```diff
@@ -54,6 +56,7 @@ export abstract class AbstractResourcePreferenceProvider extends PreferenceProvi
     protected async init(): Promise<void> {
         const uri = this.getUri();
         this.toDispose.push(Disposable.create(() => this.loading.reject(new Error(`preference provider for '${uri}' was disposed`))));
+        await this.readPreferencesFromFile();
```

Comment:
I suspect this is going to end up sending a ton of 'preferenceChanged' events. Can we prevent that? Since we haven't read the prefs yet, how could they already have changed?

Reply:
This is consistent with the current behavior, since previously the initial call to `readPreferences` emitted events. If we're willing to take a hard line and say that until the `PreferenceService` declares that it's `.ready`, users shouldn't be doing anything with it, then I think it makes sense to suppress these events (either here or on the level of the `PreferenceService` where they all get aggregated) until the `.ready` promise resolves. I'm not sure that even all framework code is on board with that approach, though, and even Ákos' code wasn't using the `.ready` promise to begin with, so it doesn't appear to be a widely-known feature.

Reply:
I've created an issue for this question [here](https://github.com/eclipse-theia/theia/issues/9367).